test/cases/oscap.sh

```diff
@@ -0,0 +1,497 @@
+groups = ["wheel"]
+EOF
+
+greenprint "📄 hardened blueprint"
+cat "${BLUEPRINT_FILE}"
+
+# Prepare the blueprint for the compose.
+greenprint "📋 Preparing hardened blueprint"
+sudo composer-cli blueprints push "${BLUEPRINT_FILE}"
+sudo composer-cli blueprints depsolve hardened
+
+# Build hardened image.
+build_image hardened qcow2
+
+# Download the image
+greenprint "📥 Downloading the image"
+sudo composer-cli compose image "${COMPOSE_ID}" > /dev/null
+IMAGE_FILENAME="${COMPOSE_ID}-disk.qcow2"
+
+greenprint "🖥 Create qcow2 file for virt install"
+sudo cp "${IMAGE_FILENAME}" "/var/lib/libvirt/images/${IMAGE_KEY}.qcow2"
+LIBVIRT_IMAGE_PATH=/var/lib/libvirt/images/${IMAGE_KEY}.qcow2
+sudo qemu-img resize "${LIBVIRT_IMAGE_PATH}" 20G
+
+# Ensure SELinux is happy with our new images.
+greenprint "👿 Running restorecon on image directory"
+sudo restorecon -Rv /var/lib/libvirt/images/
+
+greenprint "💿 Creating a cloud-init ISO"
+gen_iso
+
+greenprint "📋 Install hardened image"
+VIRSH_DOMAIN="${IMAGE_KEY}-hardened"
+sudo virt-install --name="${VIRSH_DOMAIN}"\
+                  --disk "${LIBVIRT_IMAGE_PATH}",format=qcow2 \
+                  --disk path="${CLOUD_INIT_PATH}",device=cdrom \
+                  --ram 3072 \
+                  --vcpus 2 \
+                  --network network=integration,mac=34:49:22:B0:83:30 \
+                  --os-type linux \
+                  --os-variant "${OS_VARIANT}" \
+                  --import \
+                  --nographics \
+                  --noautoconsole \
+                  --wait=15 \
+                  --noreboot
+
+# Installation can get stuck, destroying VM helps
+# See https://github.com/osbuild/osbuild-composer/issues/2413
+if [[ $(sudo virsh domstate "${VIRSH_DOMAIN}") == "running" ]]; then
+    sudo virsh destroy "${VIRSH_DOMAIN}"
+fi
+
+# Start VM.
+greenprint "💻 Start hardened VM"
+sudo virsh start "${VIRSH_DOMAIN}"
+
+# Check for ssh ready to go.
+greenprint "🛃 Checking for SSH is ready to go"
+wait_for_ssh_up "${HTTP_GUEST_ADDRESS}"
+
+greenprint "🔒 Running oscap scanner"
+scan_vm "${HTTP_GUEST_ADDRESS}"
+
+greenprint "📗 Checking oscap score"
+HARDENED_SCORE=$(ssh -q "${SSH_OPTIONS[@]}" -i "${SSH_KEY}" "${SSH_USER}@${HTTP_GUEST_ADDRESS}" \
+  sudo cat results.xml \
+  | xmlstarlet sel -N x="http://checklists.nist.gov/xccdf/1.2" -t -v "//x:score"
+)
+
+greenprint "📗 Checking for failed rules"
+SEVERITY=$(ssh -q "${SSH_OPTIONS[@]}" -i "${SSH_KEY}" "${SSH_USER}@${HTTP_GUEST_ADDRESS}" \
+  sudo cat results.xml \
+  | xmlstarlet sel -N x="http://checklists.nist.gov/xccdf/1.2" -t -v "//x:rule-result[@severity='high']" \
+  | grep -c fail
+)
+
+# Clean up HARDENED VM
+greenprint "🧹 Clean up VM"
+clean_up
+
+# Remomve tmp dir.
+sudo rm -rf "${TEMPDIR}"
+
+greenprint "🎏 Checking for test result"
+echo "Baseline score: ${BASELINE_SCORE}%"
+echo "Hardened score: ${HARDENED_SCORE}%"
+
+# compare floating point numbers
+if python3 -c "exit(${HARDENED_SCORE} > ${BASELINE_SCORE})"; then
```

Comment:
So I guess a tiny bit better is still ok?

Reply:
Yeah I guess this can be updated in a follow up - I figured with checking the rest results that had high severity this was an added bonus. We could maybe improve this by checking that it's at least x% better or maybe remove the baseline and check that the score is greater than like 80/90%